flybywiresim/a32nx
# feat(atsu): DCL, Oceanic clearance and automated ATIS update requests

## Summary of Changes
This PR activates DCL and Oceanic request of the ATC COM - pages.
Additionally is the auto-update function of the ATC Atis activate.

ATC ATISuses the same data as the AOC ATIS

## Screenshots (if necessary)



### ATIS MENU
![grafik](https://user-images.githubusercontent.com/64070348/150712477-297d687f-4fb3-4e2d-a43d-06483791766d.png)
Integration of ATIS interpretation

### ATIS Auto Update
![grafik](https://user-images.githubusercontent.com/64070348/150712511-48c32e20-3077-4fcd-a50b-201472b9fad0.png)
Checks the ATIS every ten minutes and sends a "NEW ATIS" status to the MCDU's scratchpad

### ATIS reader
![grafik](https://user-images.githubusercontent.com/64070348/150712586-6d4e3ee4-a078-49d5-8010-2c477f6a146b.png)

### DCL
![grafik](https://user-images.githubusercontent.com/64070348/150866644-9840b746-ab88-463c-95b0-04c03d242319.png)

### OCL
![grafik](https://user-images.githubusercontent.com/64070348/150866719-ea2ee3e5-c5e1-40d3-bc00-6090c9e49657.png)

### Example: DCDU with OCL
![grafik](https://user-images.githubusercontent.com/64070348/150866800-136353ee-cb66-4c0a-8fbd-bafa95694286.png)

## References

A320 FCOM documentation and used "MSN 7560" as a reference.




## Additional context



Discord username (if different from GitHub): Sven [de en]

## Testing instructions

- FREETEXT
  - send a freetext to A32NX with small letters.
  - that received message shows capital letters in the MCDU

- D-ATIS (ATC COMM -> Page 2 -> ATIS)
  - file a flightplan with airports for provided ATIS
    - three airports are shown if ALTN is set
    - fourth airport can be defined
      - format defined in [5]
    - REQ SEND is shown if airport is not defined for auto-updates
    - UPDATE SEND is shown if airport is active for auto-updates, but no ATIS is stored
    - UPDATE CANCEL is shown if airport is active for auto-updates, and ATIS is stored
  - If update failes: "NO ATIS RECV TRY LATER"
  - if update received: "NEW ATIS: READ AGAIN"
  - auto update is disabled if FROM/TO or ALTN are changed on INIT A
  - received ATIS can be read via LSK
  - print to AUTO prints all new received ATIS reports

- Depart Req (ATC COMM -> Page 2 -> DEPART REQ)
  - empty flightplan shows "------" in the callsign field
  - if ATIS available, the ATIS field is filled automatically
  - required information to send message to DCDU:
    - loggon on to ATC or station
    - callsign
    - from/to
    - ATIS
  - test if PDC is sent to ATC

- Oceanic Req (ATC COMM -> Page 2 -> OCEANIC REQ)
  - empty flightplan shows "------" in the callsign field
  - no connection to ATC shows "----" in OCEAN ATC entry, otherwise the ATC callsign
  - entry point is matched against FPL, but allows additional waypoints
    - accepted formats in [1]
  - if entry point is in FPL, the AT TIME is filled automatically
    - accepted formats in [2]
  - REQ mach accepts [3]
  - Req FL accepts [4]
  - required information to send message to DCDU:
    - OCEAN ATC
    - entry point
    - at time
    - req mach
    - req FL
    - callsign
  - Test if OCL is sent to ATC

- ATC MSG RECORD
  - default: NO MESSAGES
  - overview shows the first line of the messages
  - MSG RECORD ERASE switches to ERASE MSG RECORD CONFIRM

[1]
![grafik](https://user-images.githubusercontent.com/64070348/153026452-70e624e2-fb91-44da-a9a2-7868cc2459a6.png)

[2]
![grafik](https://user-images.githubusercontent.com/64070348/153026529-9385c15c-c136-4907-9dc7-c8006762d94a.png)

[3]
![grafik](https://user-images.githubusercontent.com/64070348/153026805-3405b59f-ae0c-4e66-95eb-31bdb671557f.png)

[4]
![grafik](https://user-images.githubusercontent.com/64070348/153029125-6396a81a-8d69-42b2-a57f-40a678cbaa58.png)

[5]
![grafik](https://user-images.githubusercontent.com/64070348/153773267-7dd6eaba-5119-4424-8b04-099e796b6ec4.png)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
